assignments/lambda-classes.js

```diff
@@ -1 +1,189 @@
 // CODE here for your Lambda Classes
+class Person {
+    constructor(person) {
+        this.name = person.name;
+        this.age = person.age;
+        this.location = person.location;
+        this.gender = person.gender;
+    }
+
+    speak() {
+        return `Hello my name is ${this.name}, I am from ${this.location}`;
+    }
+}
+
+class Instructor extends Person {
+    constructor(instructor) {
+        const person = {
+            name: instructor.name,
+            age: instructor.age,
+            location: instructor.age,
+            gender: instructor.gender
+        }
+        super(person);
+        this.specialty = instructor.specialty;
+        this.favLanguage = instructor.favLanguage;
+        this.catchPhrase = instructor.catchPhrase;
+    }
+
+    demo(subject) {
+        return `Today we are learning about ${subject}`;
+    }
+
+    grade(student, subject) {
+        return `${student.name} receives a perfect score on ${subject}`;
+    }
+
+    random(student) {
+        let max = 100 - student.grade;
+        let min = 0 - student.grade;
+        let rand = Math.floor(Math.random() * (max - min + 1)) + min;
+        return student.grade = student.grade + rand;
+    }
+}
+
+class Student extends Person{
+    constructor(student) {
+        const person = {
+            name: student.name,
+            age: student.age,
+            location: student.age,
+            gender: student.gender
+        }
+        super(person);
+        this.previousBackground = student.previousBackground;
+        this.className = student.className;
+        this.favSubjects = student.favSubjects;
+        this.grade = 90;
+    }
+
+    listSubjects() {
+        this.favSubjects.map(subject => console.log(subject));
```

Comment:
You get undefined when you log the function to the console.
You could want to refactor this function to return a string with all the values of your array.